server/main.py

```diff
@@ -2,19 +2,20 @@
 import sys
 from pathlib import Path
 
-# TODO: consolidate duplication of this block of code (seed_initial_data and conftest)
+import uvicorn
+from fastapi import FastAPI
+from fastapi.middleware.cors import CORSMiddleware
+
+# TODO: Make DRY (used in seed_initial_data and conftest)
 # if __name__ == "__main__" and server folder is not on PYTHONPATH
 server_fpath = str(Path(os.path.join(os.path.dirname(__file__))).parent)
 if server_fpath not in sys.path:
     sys.path.append(server_fpath)
 
-import uvicorn
-from fastapi import FastAPI
-from fastapi.middleware.cors import CORSMiddleware
 
-from server.api.v1.api import api_router
-from server.core.constants import API_PREFIX
-from server.db.database import db
+from server.api.v1.api import api_router  # noqa: E402
+from server.core.constants import API_PREFIX  # noqa: E402
+from server.db.database import db  # noqa: E402
```

Comment:
What is E402? 402 error code?

Reply:
noqa (stands for **N** **O** **Q**uality **A**ssurance). It's used in Python code to ignore [PEP8](https://www.python.org/dev/peps/pep-0008/) warnings. 

Those codes refer to specific linting rules. For example, E402 means "Module level import not at top of file".

Reply:
So basically you marked those comments as showing the changes you made. You moved the imports to the top of the file right?

Reply:
I didn't move those imports up. This reflects that I've made an exception to the PEP8 convention intentionally.

Reply:
But y tho?

Reply:
**Why did I not move the imports up?**

I did not move these imports up because [this block of code must be executed first](https://github.com/clairBuoyant/server/pull/40/files#diff-c16fbf0c6f7b90a46b94b36f88893c2d174476088608841f7254afba0e81373dR11-R14). This code was my hacky solution to solving python pathing issues; do you remember encountering server module error not found before?

One of the solutions to that problem is adding `PYTHONPATH=$PYTHONPATH:.` to your systems environment variables. However, in the interest of simplifying the development experience, I didn't want that to be a requirement. Hence, that block of code. It helps Python find the server module before importing stuff from server.

**Why am I using noqa?**

We added stuff like `pre-commit` and linting to standardize our code. What I did is not standard practice, so I wanted to note this as an exception by adding the `# noqa` tag. This allows these lines to pass the lint checks; the linter will ignore lines that have `# noqa` written on them.

I specified the code (e.g., `# noqa: E401`) so that the linter only makes one exception – in this case, an exception to the PEP8 rule that all import statements should be on top before any code. 

The specific codes should inform us as developers what's going on without having to write explanatory comments. In other words, it should signal that an exception was made for a particular reason – the reason is reflected by the code.

Does that make sense?

Reply:
Yes everything makes sense. I was reffering to the former answer and not the latter, but it's good to know about the latter as well.